records/generator.go

```diff
@@ -124,15 +127,19 @@ type EnumerationData struct {
 }
 
 // NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
-func NewRecordGenerator(httpClient *http.Client) *RecordGenerator {
-	enumData := EnumerationData{
-		Frameworks: []*EnumerableFramework{},
+func NewRecordGenerator(config Config) *RecordGenerator {
+	scheme, tlsClientConfig := httpcli.TLSConfig(config.MesosHTTPSOn, config.caPool)
+	return &RecordGenerator{
+		httpClient: httpcli.New(
+			config.iamConfig,
+			httpcli.Transport(&http.Transport{
+				DisableKeepAlives:   true, // Mesos master doesn't implement defensive HTTP
+				MaxIdleConnsPerHost: 2,
```

Comment:
we don't want a new pool on reload. i submitted a PR to fix this

On Tue, Jun 21, 2016 at 9:00 PM, Sargun Dhillon <notifications@github.com>
wrote:

> In records/generator.go
> <https://github.com/mesosphere/mesos-dns/pull/443#discussion_r67979493>:
>
> > @@ -124,15 +127,19 @@ type EnumerationData struct {
> >  }
> >
> >  // NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
> > -func NewRecordGenerator(httpClient *http.Client) *RecordGenerator {
> > -	enumData := EnumerationData{
> > -		Frameworks: []*EnumerableFramework{},
> > +func NewRecordGenerator(config Config) *RecordGenerator {
> > +	scheme, tlsClientConfig := httpcli.TLSConfig(config.MesosHTTPSOn, config.caPool)
> > +	return &RecordGenerator{
> > +		httpClient: httpcli.New(
> > +			config.iamConfig,
> > +			httpcli.Transport(&http.Transport{
> > +				DisableKeepAlives:   true, // Mesos master doesn't implement defensive HTTP
> > +				MaxIdleConnsPerHost: 2,
>
> If we're creating a new pool on every failover, any reason to not make
> this 1? or 0?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/443/files/00e2c3a5bbcc291550b5d0707072d2359f991fbb#r67979493>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACPVLFnYXoW7sktmxlRLsFSws5RCZujoks5qOIk3gaJpZM4I6PXJ>
> .
>